### ISSUE TITLE
Height as input on ESPhome

### ISSUE THREAD

**Issue by Steff1990** [richard-kramer/exeta-desk-controller#1]
Hi Richard, i really love this project . 
This is what kam looking for. The only thing what i  miss is the possibility to enter the height as Input. 
so you think, this is possible?

Thank you

**Comment by richard-kramer** [comment]
It might be possible. Not having this option wasn't a big problem for me, so I never really explored it. I really only ever set two of the three memory positions and alternate between them.

I guess, the ESP would just have to emulate holding the "up" or "down" button until the desired height is reached. Maybe I can take the time and look into implementing this in the next few days.

**Comment by Steff1990** [comment]
> It might be possible. Not having this option wasn't a big problem for me, so I never really explored it. I really only ever set two of the three memory positions and alternate between them.
> 
> I guess, the ESP would just have to emulate holding the "up" or "down" button until the desired height is reached. Maybe I can take the time and look into implementing this in the next few days.

I would really love this. Sorry i‘m really new to Home Assistant and so i’m not able to do it at my own…
Best regards from Germany ;)

**Comment by richard-kramer** [comment]
I implemented a solution which also includes a cover component. Unfortunately, it's a bit unprecise. You may read [the disclaimer concerning these entities](https://github.com/richard-kramer/exeta-desk-controller?tab=readme-ov-file#remote-control).

@Steff1990 Let me know if this works for you or leaves something to desire.

**Comment by Steff1990** [comment]
> I implemented a solution which also includes a cover component. Unfortunately, it's a bit unprecise. You may read [the disclaimer concerning these entities](https://github.com/richard-kramer/exeta-desk-controller?tab=readme-ov-file#remote-control).
> 
> @Steff1990 Let me know if this works for you or leaves something to desire.

Thank you, World really well.
So the only point, but this is also at the Last Version. My Display only shows 00.0.
If I remove it even reset my esp, it Shows the right value again, intim the esp Starts up.

do you know this issue?

**Comment by richard-kramer** [comment]
Actually never had this. Maybe they changed something in the communication protocol on newer models? Just for reference, my control panel looks like this, but without the big logo:

![image](https://github.com/user-attachments/assets/fa06f6d1-fff5-4b33-84b2-afa8c69bfe11)

As far as I analysed the communication between the desk controller and the control panel, the height display should get its value directly from the UART RX pin (that I numbered as PIN4). The ESP should only read this signal and not modify it in any way.

Do you maybe use an external power supply with the ESP? It might also be possible, the ESP sends some signal via one of the pins when starting up. Did you change something on the pinout or are you using the exact same hardware?

**Comment by Steff1990** [comment]
> Actually never had this. Maybe they changed something in the communication protocol on newer models? Just for reference, my control panel looks like this, but without the big logo:
> 
> ![image](https://private-user-images.githubusercontent.com/34233066/386768673-fa06f6d1-fff5-4b33-84b2-afa8c69bfe11.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE4MzgxOTEsIm5iZiI6MTczMTgzNzg5MSwicGF0aCI6Ii8zNDIzMzA2Ni8zODY3Njg2NzMtZmEwNmY2ZDEtZmZmNS00YjMzLTg0YjItYWZhOGM2OWJmZTExLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTE3VDEwMDQ1MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWZlMWVmM2VlYWM0NmZkNGZhOWU4NWIxZDk3ZTFmNGRjODA5ZmI2M2I2NjlmZDUyZTUxOThjYjhkMmY1ODY0YWQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.VpdYtt7PcXCX6B0U-cq53rBE-4Yvb1R5QTTWI8bonC8)
> 
> As far as I analysed the communication between the desk controller and the control panel, the height display should get its value directly from the UART RX pin (that I numbered as PIN4). The ESP should only read this signal and not modify it in any way.
> 
> Do you maybe use an external power supply with the ESP? It might also be possible, the ESP sends some signal via one of the pins when starting up. Did you change something on the pinout or are you using the exact same hardware?

Hello Richard, 
First of all, I wanted to thank you once again for the great integration. Everything is working perfectly now.
I also think that the height setting is very accurate with your adjustments ;)
Now to my error:
I looked at the code again:
uart:
```
  id: uart_bus
  tx_pin: D6
  rx_pin: D5
  baud_rate: 9600
```
Here you have specified that D5 is the RX pin.
So I used D5 instead of D6 as in your fritzing picture and what can I say, it works perfectly. 
Only one last error for Auto Desk Height:
"error if you don't pass an entity, you mast pass 2 time/duration properties"
Many thanks for your efforts.
Have a nice weekend.

**Comment by richard-kramer** [comment]
I checked the wiring diagram and my photos and you're absolutely right. I wired the PIN4 (RX) from the controller board to **D5** instead of **D6** myself and must have made a mistake when creating the diagram and writing the description. Of cause, RX has to be wired as RX.

I'll update the documentation, switching the `rx_pin` and `tx_pin` in the ESP code should be enough.

Thanks for this finding.
